code/game/gamemodes/traitor/traitor.dm

```diff
@@ -87,9 +87,16 @@
 		for(var/datum/antagonist/traitor/traitor_datum in traitor_mind.antag_datums)
 			for(var/datum/objective/objective in traitor_datum.objective_holder.objectives)
 				objective.delayed_objective = FALSE
+				objective.target = null
+
 				var/datum/objective/steal/possible_steal_objective = objective
-				if(istype(possible_steal_objective) && possible_steal_objective.steal_target.special_equipment)
-					possible_steal_objective.give_kit(possible_steal_objective.steal_target.special_equipment)
+				if(istype(possible_steal_objective))
+					possible_steal_objective.steal_target = null
```

Comment:
Make a helper like `found_target` here, but for setting the target please

Reply:
I added `force_reset_target`, is that what you meant?